materialize-sql/type_mapping.go

```diff
@@ -387,6 +380,30 @@ type StringTypeMapper struct {
 var _ TypeMapper = StringTypeMapper{}
 
 func (m StringTypeMapper) MapType(p *Projection) (MappedType, error) {
+	// TODO(whb): This bit of hackery is to provide backwards compatibility for materializations
```

Comment:
This section was moved out of `table_mapping.go`, since I think it makes more sense here. It is still necessary to have this because of one postgres materialization we have running.